pkg/constants/constants.go

```diff
@@ -14,14 +14,24 @@
 
 package constants
 
+import "os"
+
 const (
 	CurrentAPIVersion = "krew.googlecontainertools.github.com/v1alpha2"
 	PluginKind        = "Plugin"
 	ManifestExtension = ".yaml"
 	KrewPluginName    = "krew" // plugin name of krew itself
 
-	// DefaultIndexURI points to the upstream index.
-	DefaultIndexURI = "https://github.com/kubernetes-sigs/krew-index.git"
 	// DefaultIndexName is a magic string that's used for a plugin name specified without an index.
 	DefaultIndexName = "default"
 )
+
+// DefaultIndexURI points to the upstream index.
+var DefaultIndexURI = getDefaultIndexURI()
+
+func getDefaultIndexURI() string {
+	if uri := os.Getenv("KREW_DEFAULT_INDEX_URI"); uri != "" {
+		return uri
+	}
+	return "https://github.com/kubernetes-sigs/krew-index.git"
+}
```

Comment:
I'd suggest to use this pattern:

```suggestion
const defaultIndexURI = "https://github.com/kubernetes-sigs/krew-index.git"
var DefaultIndexURI string = defaultIndexURI

func init() {
	if uri := os.Getenv("KREW_DEFAULT_INDEX_URI"); uri != "" {
		DefaultIndexURI = uri
	}
}
```

Then you can unit-test this logic by explicitly calling `init()` in your test. However, I think this function is so simple that it does necessarily not need one.